### ISSUE TITLE
Assembler highlighting/navigation not working for additional file extensions (.as)

### ISSUE THREAD

**Issue by doubleell** [maziac/asm-code-lens#67]
I have added .as as an additional file extension for assembler but none of the syntax highlighting or navigation is working, the text appears as standard text.

**Comment by maziac** [comment]
Where have you added that?

**Comment by doubleell** [comment]
In the settings section for Include Files;
**/*.{asm,inc,s,a80,as}

![image](https://user-images.githubusercontent.com/36256378/157621730-c4615a56-5e6e-49ad-aecc-f9128191554d.png)

**Comment by maziac** [comment]
Additionally you need to set the vscode file association.
Open the as file, then click in right bottom corner (instead of 'Assembler file' there might be something different for you):
![Bildschirmfoto 2022-03-10 um 19 58 35](https://user-images.githubusercontent.com/3081973/157735427-44cc3ca2-116c-443b-b256-6e1b96bf5a9a.jpg)

Choose 'Configure file association':
![Bildschirmfoto 2022-03-10 um 19 57 53](https://user-images.githubusercontent.com/3081973/157735311-50c6859e-6725-4849-a4e9-7b17c04268d3.jpg)

And choose 'Assembler file':
![Bildschirmfoto 2022-03-10 um 19 59 15](https://user-images.githubusercontent.com/3081973/157735534-704e7174-0ffa-469a-8f53-71ffd463646c.jpg)

Reason behind this is that the syntax highlighting and the other features require different settings.

**Comment by doubleell** [comment]
Hi,
I’ve used the extract method using regex and it searches great. When I fill
out the 2nd field for output I’m using $1 etc, but how do I insert new
lines?

I’m trying to format an assembler function as c

Eg. void $1(void) {}

But I wanted new lines inserted?

Regards,
Allan.

**Comment by maziac** [comment]
Not sure what you are talking about. Is this a new topic or is it still about syntax highlighting?
Did you forked/manipulated the code? What "extract method" it is about?

**Comment by doubleell** [comment]
I guess I’m asking if it’s possible to implement, I’m using the extract
text to new file but for the second input box I wanted to insert new lines.

**Comment by maziac** [comment]
You need to give me more context. I actually have no idea what you are trying to achieve.

**Comment by doubleell** [comment]
Hello,
Sorry I had go

**Comment by doubleell** [comment]
Hello,
Sorry I had got your extension mixed up with someone else's. I made the
change as you suggested and the assembler file highlighting is great
thankyou. Apologies for this....

Regards,
Allan.

**Comment by maziac** [comment]
No problem.

**Comment by doubleell** [comment]
I thought I already explained, I wanted new lines in the new file. $1 is
the capture group from the first input box, this would be the function name
used in the second input box. The first input box uses a regex to extract a
function name from an assembler file.

At the moment the second input box of void $1(void) {};
is a single line, I tried void $1(void) \n {\n };\n
That just puts the literal characters \n in the output.

I wanted the output below, where $1 is the function name;
void $1(void)
{
};

Regards,
Allan.
